### ISSUE TITLE
Error after upgrading to v9: An unhandled exception occurred: Job name "..getProjectMetadata" does not exist.

### ISSUE THREAD

**Issue by cah-alexsukhodolsky** [angular/angular-cli#16862]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [x] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Yes. It worked in the latest version of Angular 8

### Description

Running I upgraded my project to Angular 9 by running `ng update @angular/cli @angular/core`. After completion I tried running `ng serve` but got the error found below.

## 🔬 Minimal Reproduction
See above.

## 🔥 Exception or Error
<pre><code>

An unhandled exception occurred: Job name "..getProjectMetadata" does not exist.
See "/private/var/folders/lk/wvqwdphx6rl56j3cl3yqzkxm0000gn/T/ng-Q9Si6x/angular-errors.log" for further details.

</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 9.0.0
Node: 13.3.0
OS: darwin x64

Angular: 9.0.0
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, material, platform-browser
... platform-browser-dynamic, router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.900.0
@angular-devkit/build-angular      0.900.0
@angular-devkit/build-optimizer    0.900.0
@angular-devkit/build-webpack      0.900.0
@angular-devkit/core               9.0.0
@angular-devkit/schematics         7.3.9
@angular/flex-layout               9.0.0-beta.29
@angular/material-moment-adapter   8.2.3
@ngtools/webpack                   9.0.0
@nguniversal/express-engine        7.1.1
@schematics/angular                7.3.9
@schematics/update                 0.900.0
rxjs                               6.5.4
typescript                         3.7.5
webpack                            4.41.2
</code></pre>

**Comment by clydin** [comment]
Can you please provide the contents of the project's `package.json`?

**Comment by cah-alexsukhodolsky** [comment]
<pre><code>
"dependencies": {
    "@agm/core": "^1.0.0-beta.7",
    "@alyle/ui": "^2.9.7",
    "@angular-material-extensions/google-maps-autocomplete": "^2.0.0",
    "@angular/animations": "^9.0.0",
    "@angular/cdk": "^9.0.0",
    "@angular/common": "^9.0.0",
    "@angular/compiler": "^9.0.0",
    "@angular/core": "^9.0.0",
    "@angular/flex-layout": "9.0.0-beta.29",
    "@angular/forms": "^9.0.0",
    "@angular/material": "^9.0.0",
    "@angular/material-moment-adapter": "^8.2.3",
    "@angular/platform-browser": "^9.0.0",
    "@angular/platform-browser-dynamic": "^9.0.0",
    "@angular/router": "^9.0.0",
    "@fortawesome/angular-fontawesome": "^0.5.0",
    "@fortawesome/fontawesome-pro": "^5.10.2",
    "@fortawesome/fontawesome-svg-core": "^1.2.25",
    "@mapbox/mapbox-gl-geocoder": "^4.5.1",
    "@ng-toolkit/serverless": "^7.1.2",
    "@ng-toolkit/universal": "^7.1.2",
    "@ngrx/effects": "^8.3.0",
    "@ngrx/router-store": "^8.3.0",
    "@ngrx/store": "^8.3.0",
    "@ngx-translate/core": "^11.0.1",
    "@ngx-translate/http-loader": "^4.0.0",
    "@sentry/browser": "^5.10.2",
    "angular-froala-wysiwyg": "^3.0.6",
    "change-case": "^4.1.1",
    "chart.js": "^2.8.0",
    "chroma-js": "^2.0.6",
    "codemirror": "^5.49.2",
    "core-js": "^3.2.1",
    "countup.js": "^2.0.4",
    "font-awesome": "^4.7.0",
    "fullcalendar": "^3.10.1",
    "fuse.js": "^3.4.5",
    "hammerjs": "^2.0.8",
    "jquery": "^3.4.1",
    "jwt-decode": "^2.2.0",
    "libphonenumber-js": "^1.7.23",
    "lodash": "^4.17.15",
    "mapbox-gl": "1.6.1",
    "moment": "2.24.0",
    "net": "^1.0.2",
    "ng-intercom": "^8.0.2",
    "ng2-charts": "^2.3.0",
    "ng2-tooltip-directive": "^2.8.17",
    "ngx-joyride": "^2.2.10",
    "ngx-mapbox-gl": "^4.3.0",
    "ngx-mask": "^8.0.8",
    "ngx-material-file-input": "^2.0.0",
    "ngx-material-timepicker": "^4.0.2",
    "ngx-perfect-scrollbar": "^8.0.0",
    "ngx-webstorage": "4.0.1",
    "resize-observer-polyfill": "^1.5.1",
    "rxjs": "^6.5.3",
    "sockjs-client": "^1.4.0",
    "stompjs": "^2.3.3",
    "tributejs": "^4.1.1",
    "tslib": "1.10.0",
    "typescript-string-operations": "^1.3.1",
    "webstomp-client": "1.2.6",
    "zone.js": "~0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.900.0",
    "@angular/cli": "9.0.0",
    "@angular/compiler-cli": "^9.0.0",
    "@angular/language-service": "~9.0.0",
    "@ngrx/store-devtools": "^8.6.0",
    "@types/dotdotdot": "^1.6.34",
    "@types/googlemaps": "^3.39.0",
    "@types/jasmine": "^3.4.0",
    "@types/jasminewd2": "^2.0.5",
    "@types/jquery": "^3.3.31",
    "@types/mapbox-gl": "1.6.1",
    "@types/node": "12.7.4",
    "@types/shortid": "0.0.29",
    "aws-sdk": "^2.522.0",
    "codelyzer": "^5.0.1",
    "jasmine-core": "^3.3.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "^4.3.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "^2.1.0",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.0",
    "karma-phantomjs-launcher": "^1.0.4",
    "phantomjs-prebuilt": "^2.1.16",
    "protractor": "^5.4.1",
    "rimraf": "3.0.0",
    "ts-node": "~8.3.0",
    "tslint": "5.19.0",
    "typescript": "3.7.5"
  }
</code></pre>

**Comment by clydin** [comment]
This error is caused by an old version (< 9.0.0) of `@angular-devkit/architect` being used.  It doesn't appear to be present in the dependencies.  However, one of the third-party dependencies may be including it.  Are there any third-party builders being used in the project?
From the version output, the following also doesn't appear correct and should both be `9.0.0`:
```
@angular-devkit/schematics         7.3.9
@schematics/angular                7.3.9
```
This also may be due to one or more of the third-party dependencies not being fully compatible with newer versions of Angular.

One thing that can be tried in situations like this is to completely remove the `node_modules` directory along with any lock files (package-lock.json or yarn.lock) and reinstalling.

**Comment by cah-alexsukhodolsky** [comment]
@clydin Deleting the `node_modules` directory and yarn.lock file fixed the issue. Thank you!

**Comment by Lightras** [comment]
In my case, we didn't try to upgrade to Angular 9, but the @angular-devkit/build-angular version in package.json became ^0.900.1, which seems not suitable for Angular 8. So what worked for me was downgrading this dependency to the fixed version we used before (0.803.8)

**Comment by AnkitEncom** [comment]
Same issue i have found in my newly created project.

npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.

How can i solve this issue.

**Comment by dlockliear** [comment]
After running ng update, I realized I hadn't updated @angular/cli (8.0.6)  -> 9.0.2... ng update @anuglar/cli worked for me.

**Comment by AndreKoepke** [comment]
> In my case, we didn't try to upgrade to Angular 9, but the @angular-devkit/build-angular version in package.json became ^0.900.1, which seems not suitable for Angular 8. So what worked for me was downgrading this dependency to the fixed version we used before (0.803.8)
> 
> And yes, I've tried deleting and reinstalling the node_modules directory - this didn't help

That was the only solution to solve this issue for us. For others: Check the latest version of `@angular-devkit/build-angular` [@0.803.x here](https://www.npmjs.com/package/@angular-devkit/build-angular).  
Best way to archive this: `"@angular-devkit/build-angular": "0.803"`

**Comment by OceansideBill** [comment]
It appears that the latest version of teh @angular-devkit/build-angular (currently 0.803.21) has a security vulnerability which is causing an "npm audit fix" to update it to 0.900.*, which isn't a compatible change.  Keeping this on the older version and accepting the High security issue appears to be the current work around.

**Comment by lhopkins-tucasi** [comment]
> It appears that the some versions of the v8 @angular-devkit/build-angular (I was on 0.803.21) has a security vulnerability which is causing an "npm audit fix" to update it to 0.900.*, which isn't a compatible change. Need to move to 0.803.25 instead of the npm audit fix solution and both the error and the security warning are resolved.

Just happened to me, `npm audit fix` automatically upgraded @angular-devkit/build-angular to v9, which doesn't work on a Angular v8 project. Cheers @OceansideBill, manually upgraded to 0.803.25 and serve works again
